src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java

```diff
@@ -42,35 +51,54 @@ public static void init() {
 				String titleString = title.getString();
 				if (DUNGEON_CHESTS.contains(titleString)) {
 					if (SkyblockerConfigManager.get().dungeons.dungeonChestProfit.enableProfitCalculator) {
-						ScreenEvents.afterTick(screen).register(screen_ ->
-								((ScreenAccessor) screen).setTitle(getDungeonChestProfit(genericContainerScreen.getScreenHandler(), title, titleString))
-						);
+                        Text dungeonChestProfit = getDungeonChestProfit(genericContainerScreen.getScreenHandler());
+                        if (dungeonChestProfit != null) addValueToContainer(genericContainerScreen, dungeonChestProfit, title);
 					}
 				} else if (SkyblockerConfigManager.get().uiAndVisuals.chestValue.enableChestValue && !titleString.equals("SkyBlock Menu")) {
+					boolean minion = MINION_PATTERN.matcher(title.getString().trim()).find();
 					Screens.getButtons(screen).add(ButtonWidget
 							.builder(Text.literal("$"), buttonWidget -> {
 								Screens.getButtons(screen).remove(buttonWidget);
-								ScreenEvents.afterTick(screen).register(screen_ ->
-										((ScreenAccessor) screen).setTitle(getChestValue(genericContainerScreen.getScreenHandler(), title, titleString))
-								);
+									Text chestValue = getChestValue(genericContainerScreen.getScreenHandler(), minion);
+                                    if (chestValue != null) {
+                                        addValueToContainer(genericContainerScreen, chestValue, title);
+                                    }
 							})
 							.dimensions(((HandledScreenAccessor) genericContainerScreen).getX() + ((HandledScreenAccessor) genericContainerScreen).getBackgroundWidth() - 16, ((HandledScreenAccessor) genericContainerScreen).getY() + 4, 12, 12)
-							.tooltip(Tooltip.of(Text.translatable("skyblocker.config.general.chestValue.@Tooltip")))
+							.tooltip(minion ? Tooltip.of(Text.translatable("skyblocker.config.general.minionValue.@Tooltip")) : Tooltip.of(Text.translatable("skyblocker.config.general.chestValue.@Tooltip")))
 							.build()
 					);
 				}
 			}
 		});
 	}
 
-	private static Text getDungeonChestProfit(GenericContainerScreenHandler handler, Text title, String titleString) {
+	private static void addValueToContainer(GenericContainerScreen genericContainerScreen, Text chestValue, Text title) {
```

Comment:
Can you move this metho all the way to the end (before `ScrollingTextWidget`) cause this method is called last.